### DISCUSSION TITLE
Add animation duration adjustment / allow for smooth re-toggle.

### DISCUSSION THREAD

**Discussion by nac0n** [mdbootstrap/TW-Elements#2053] [Share ideas & request features]
Hi there, Lovely library!

I have been looking into this library lately and is liking a lot of the design choices, like for example the label in an input moving up to above the field while there's text in the input-box.

I'm rather curious about features like no animation duration- adjustment for things like dropdown-menu, where I'd would like to adjust the time it takes for the menu to finish it's animation. 
I noticed that Ripple has this feature. So I'm mostly wondering; If it's a design choice or just something that's not been thought about adding? At least I've not noticed anything about it...

 I really love a lot about TW-elements so it would be a shame if this was the dealbreaker.

**Comment by nac0n** [comment]
Terribly sorry, but i now noticed the class "duration" is available for the accordion also. OP comment is a none issues. Thanks for such a great library!

**Comment by juujisai** [comment]
Hi! We are aiming to provide as many customization options as possible but I guess sometimes we miss something.

I see that dropdown doesn't have the customization for that and the value is hardcoded. I'm adding a new task to make it possible. As of the rest of components, I'm also adding a task to check what should be added and how we can make the animation customizations possible for them.

Animations from the https://tw-elements.com/docs/standard/content-styles/animations/ can be customized just be editing the timing inside the data attibute (just have to remember to add the class to the safelist). For example:
```
data-te-animation="[slide-right_5s_ease-in-out]"
```
